src/libraries/System.Runtime.Numerics/src/System/Numerics/BigInteger.cs

```diff
@@ -1084,12 +1084,10 @@ public override int GetHashCode()
             if (_bits is null)
                 return _sign;
 
-            int hash = _sign;
-            for (int iv = _bits.Length; --iv >= 0;)
-                hash = unchecked((int)CombineHash((uint)hash, _bits[iv]));
-            return hash;
-
-            static uint CombineHash(uint u1, uint u2) => ((u1 << 7) | (u1 >> 25)) ^ u2;
+            HashCode hash = default;
+            hash.Add(_sign);
+            hash.AddBytes(MemoryMarshal.AsBytes(_bits.AsSpan()));
```

Comment:
Should we reverse these so that we don't have to go through the flushing logic in AddBytes?

Reply:
In #53984 I was suggested to use `AddBytes`. Not very sure about the characteristics about them.

Reply:
Yes, the ask here is to do :
```suggestion
            hash.AddBytes(MemoryMarshal.AsBytes(_bits.AsSpan()));
            hash.Add(_sign);
```

That is, just inverse the call order as it simplifies how much work the algorithm needs to do. 

https://source.dot.net/#System.Private.CoreLib/HashCode.cs,435866171118503a shows that it has to pessimize the algorithm slightly if we've already added bytes into the hash code and haven't fit a full 16 yet. Doing the `AddBytes` first fixes that and makes everything just slightly cleaner.